anesthetic/plot.py

```diff
@@ -281,6 +308,35 @@ def fastkde_plot_1d(ax, data, *args, **kwargs):
 
     ans = ax.plot(x[i], p[i], color=color, *args, **kwargs)
     ax.set_xlim(*check_bounds(x[i], xmin, xmax), auto=True)
+
+    if fill is True:
+        lims = iso_probability_1d(x[i], p[i], levels)
+        c = [round(1-level, 2) for level in levels]
+        for j in range(len(lims)):
+            if j < len(lims) - 1:
+                if lims[j] != lims[j+1]:
+                    x_per = np.linspace(lims[j], lims[j+1], 1000)
+                    y_per = np.interp(x_per, x[i], p[i])
+                    y_max = round(y_per.max(), 2)
+                    if y_max > max(c):
+                        alpha = max(fill_alpha)
+                    else:
+                        for h in range(len(c)):
+                            if h > 0:
+                                if y_max > c[h] and \
+                                        y_max <= c[h-1]:
+                                    alpha = fill_alpha[h]
+                    if invert_alpha is True:
+                        ax.fill_between(
+                            x_per, y_per,
+                            color=fill_color,
+                            alpha=1-alpha)
+                    else:
+                        ax.fill_between(
+                            x_per, y_per,
+                            color=fill_color,
+                            alpha=alpha)
+
```

Comment:
This code is repeated below, so if it does remain, you should probably move it into a helper function to avoid code repetition.